cryotypes/image/_validators.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+
+from ._protocol import ImageProtocol
+
+if TYPE_CHECKING:
+    from typing import Literal
+
+
+def validate_image(
+    image: Any,
+    ndim: Literal[2, 3] = 3,
+    stack: bool = False,
+    coerce: bool = False,
+) -> Any:
+    """Validate an image."""
+    if not isinstance(image, ImageProtocol):
+        raise ValueError(f"{image.__class__} object does not follow the ImageProtocol")
+
+    if not hasattr(image.data, "__array__"):
+        if not coerce:
+            raise ValueError(f"{image.__class__}.data is not an ArrayLike")
+        else:
+            image.data = np.asanyarray(image.data)
+
+    total_ndim = ndim + stack
+    if image.data.ndim != total_ndim:
+        if not coerce:
+            raise ValueError(f"data must be {total_ndim}D, got {image.data.ndim}D")
+        elif image.data.ndim > total_ndim:
+            raise ValueError(
+                f"cannot coerce {image.data.ndim}D data to {total_ndim}D. "
+                "Did you set stack correctly?"
+            )
+        else:
+            image.data = np.expand_dims(
+                image.data, axis=tuple(range(total_ndim - image.data.ndim))
+            )
```

Comment:
This file is the same as the tomogram validator we already have, just with this part changed to account for `stack`